### PR TITLE
chore(kdatetimepicker): use Timeperiod from public-ui [MA-1995]

### PR DESCRIPTION
**Pull request by mihai-peteu**
# Summary
https://konghq.atlassian.net/browse/MA-1995

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Tests coverage:** test coverage was added for new features and bug fixes
* [ ] **Docs:** includes a technically accurate README

### CODE REVIEW

File: src/types/date-time-picker.ts [hunk 1]
```diff
@@ -1,20 +1,11 @@
 import type { AnyElementOf } from '@/types/utils'
-
+import type { TimePeriod } from '@kong-ui-public/analytics-utilities'
 export interface TimeRange {
```
Comment (filipgutica) [inline src/types/date-time-picker.ts:3]: Can import this too from analytics-utilities

https://github.com/Kong/public-ui-components/blob/8c720c694daf40b45947886357d1b44bb3b4647e/packages/analytics/analytics-utilities/src/types/datepicker-selection.ts

**Review by filipgutica — CHANGES_REQUESTED** [review]

File: src/types/date-time-picker.ts (hunk 1 above)
Comment (adamdehaven) [reply src/types/date-time-picker.ts:3]: No, kongponents cannot have a dependency on another kong package

File: package.json [hunk 2]
```diff
@@ -50,6 +50,7 @@
     "semantic-release": "semantic-release"
   },
   "dependencies": {
+    "@kong-ui-public/analytics-utilities": "^0.7.2",
```
Comment (adamdehaven) [inline package.json:53]: Kongponents cannot have a dependency on this package - this repo is completely isolated from other kong component libraries

**Review by adamdehaven — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (mihai-peteu) [reply package.json:53]: Understood

**Pull request closed without merging**